src/kemal/param_parser.cr

```diff
@@ -5,16 +5,35 @@ require "json"
 # context.
 alias AllParamTypes = Nil | String | Int64 | Float64 | Bool | Hash(String, JSON::Type) | Array(JSON::Type)
 
+class Kemal::ParamContainer
+  property url
+  property query
+  property body
+  property json
+
+  def initialize
+    @url = {} of String => String
+    @query = {} of String => String
+    @body = {} of String => String
+    @json = {} of String => AllParamTypes
+  end
+
+  def all
+    @url.merge(@query).merge(@body).merge(@json)
```

Comment:
I made it in the order of severity. But when I think again URL is much more important. Makes sense.